File: testers/avro/kotlin/generated-and-checked-in/com/example/events/consumer/DynamicValueConsumer.kt
package com.example.events.consumer

import com.example.events.DynamicValue
import com.example.events.header.StandardHeaders
import java.io.Closeable
import java.time.Duration
import org.apache.kafka.clients.consumer.Consumer
import org.apache.kafka.clients.consumer.ConsumerRecords

/** Type-safe consumer for dynamic-value topic */
data class DynamicValueConsumer(
  val consumer: Consumer<kotlin.String, DynamicValue>,
  val handler: DynamicValueHandler,
  val topic: kotlin.String = "dynamic-value"
) : Closeable {
  /** Close the consumer */
  override fun close() {
    consumer.close()
  }

  /** Poll for messages and dispatch to handler */
  fun poll(timeout: Duration) {
    val records: ConsumerRecords<kotlin.String, DynamicValue> = consumer.poll(timeout)
    records.forEach({ record -> val key: kotlin.String = record.key()
    val value: DynamicValue = record.value()
    val headers: StandardHeaders = StandardHeaders.fromHeaders(record.headers())
    handler.handle(key, value, headers) })
  }
}

File: testers/avro/scala/generated-and-checked-in/com/example/events/consumer/DynamicValueConsumer.scala
package com.example.events.consumer

import com.example.events.DynamicValue
import com.example.events.header.StandardHeaders
import java.lang.AutoCloseable
import java.time.Duration
import org.apache.kafka.clients.consumer.Consumer
import org.apache.kafka.clients.consumer.ConsumerRecords

/** Type-safe consumer for dynamic-value topic */
case class DynamicValueConsumer(
  consumer: Consumer[String, DynamicValue],
  handler: DynamicValueHandler,
  topic: String = "dynamic-value"
) extends AutoCloseable {
  /** Poll for messages and dispatch to handler */
  def poll(timeout: Duration): Unit = {
    val records: ConsumerRecords[String, DynamicValue] = consumer.poll(timeout)
    records.forEach(record => { val key: String = record.key; val value: DynamicValue = record.value; val headers: StandardHeaders = StandardHeaders.fromHeaders(record.headers); handler.handle(key, value, headers) })
  }

  /** Close the consumer */
  override def close: Unit = {
    consumer.close
  }
}